### ISSUE TITLE
istio-tracing (jaeger) has memory leak and crashes under load

### ISSUE THREAD

**Issue by jnativio** [istio/istio#5782]
Istio build: release-0.8-20180523-09-15 (daily)
Kube version: 1.9.2_1501
Cloud environment: IBM Cloud Kubernetes Service (IKS)
Kube Cluster: 3-node 4-core, 16GB 

Regression patrol run on this build showed that the new istio-tracing component that now is installed with the default manifest file (istio-demo.yaml) has a serious memory leak under load and uses all memory available in the node, then crashes and it is restarted.

![image](https://user-images.githubusercontent.com/18554027/40428788-179d77b2-5e6f-11e8-92c3-3269bdad5662.png)

**Comment by linsun** [comment]
Is this a new issue that exist with zipkin or only to jaeger?

**Comment by jnativio** [comment]
I observed that zipkin had the same issue when it was used with daily build release-0.8-20180519-01-13:

![image](https://user-images.githubusercontent.com/18554027/40492800-5367d3de-5f3f-11e8-87cb-e646c1309975.png)

**Comment by pavolloffay** [comment]
@jnativio does zipkin crash? IIRC it uses bounded in-memory storage so it shouldn't crash if it has enough memory.

**Comment by mandarjog** [comment]
@jnativio did we just start using zipkin in regpatrol ? Do we have any earlier data points?
Otherwise this is really a "how to productionize jaeger/zipkin" issue.
If so I will not consider this release blocking.

**Comment by jnativio** [comment]
@mandarjog  Both zipkin and jaeger got into the regpatrol run as they were included in the manifest yaml file. That happens only once for each one and I have been deleting them after the deployment as they interfere with the results. Besides the memory leak, they use a significant amount of CPU.

**Comment by mandarjog** [comment]
Ok, I will remove this from release blocking then.
@hklai @ldemailly This is not a regression. Both zipkin and jaeger have trouble with 100% tracing and the amount of traffic we generate in these test when used with default configuration.

I think we should improve the default configuration, create a new config for perf tests that includes trace sampling.

**Comment by jnativio** [comment]
With Istio 0.8 daily build 'release-0.8-20180531-09-15', the memory usage by istio-tracing seems to be limited to 1 GB:

![image](https://user-images.githubusercontent.com/18554027/40785195-da3ff406-64b5-11e8-8123-f8efbe5402aa.png)

**Comment by mandarjog** [comment]
Awesome.
Thanks @objectiser https://github.com/istio/istio/pull/5873
